lib/chef/knife/exec.rb

```diff
@@ -40,7 +40,7 @@ class Chef::Knife::Exec < Chef::Knife
   end
 
   def run
-    config[:script_path] ||= Array(Chef::Config[:script_path])
+    config[:script_path] = Array(config[:script_path] || Chef::Config[:script_path])
```

Comment:
due to the merging of configs, i think you can just use `config[:script_path]` here, `Chef::Config[:script_path]` should already be in there.

Reply:
As I can verify for now we required both
1. when `knife[:script_path]` is set
```
knife[:script_path] =     "/chef/scripts-path"

=> puts config[:script_path]
/chef/scripts-path

=> puts Chef::Config[:script_path]
[]
```
2. when `script_path` is provided
```
script_path    "/chef/scripts-path"

=> puts config[:script_path]
nil
=> puts Chef::Config[:script_path]
["/chef/scripts-path"]
```
let me know if I am missing something? Thanks!

Reply:
Ah right, config[:script_path] = Chef::Config[:knife][:script_path] and not Chef::Config[:script_path] and the latter is not in the knife namespace because this option is likely very old.

We should probably audit everything in knife for Chef::Config references that aren't in the knife namespace.